### DISCUSSION TITLE
<DatePicker fixOnBlur={false} allowFreeInput minDate={...} /> should not set the min/max date

### DISCUSSION THREAD

**Discussion by kevinbeal** [mantinedev/mantine#1489] [Feature requests]
When you set a `minDate` and have `allowFreeInput` it will set the date to the `minDate` when the value is before that date. The same behavior works in the other direction for `maxDate`. This makes sense and is a good feature (thank you).

My assumption was when setting the `fixOnBlur` prop to `false` that this behavior would be canceled. My assumption was incorrect and it only stops making an invalid date into the last valid one. I can imagine there is a good reason for this, and am open to counter argument, but I think it should work as it currently does but with the addition of stopping the min/max date value change as well.

"Fix" in my mind sounds like "make into a value the input will accept as valid". From that perspective, changing the date to the min date when before it is a "fix". So, `fixOnBlur={false}` should then also stop setting it to the `minDate`.

Of course, this is just my opinion, but in the snippet below, the condition on line 197 should be `allowFreeInput && fixOnBlur`.
![image](https://user-images.githubusercontent.com/855252/169619316-2775c27a-d8f6-4a9d-a697-36911fdea2c4.png)

**Comment by rtivital** [comment]
I do not understand the issue, can you provide an example of user input and result value?

**Reply by kevinbeal** [reply]
Yeah. When I have the min date set to 2022-02-02 and I type 2022-02-01 the date will change from 2022-02-01 to 2022-02-02 when I blur the field. 

If 'fixInBlur' is false, it still does that when I expect it not to.

**Reply by rtivital** [reply]
It is how input works with minDate, `fixOnBlur` handles only formatting issues. Anyway, allowing invalid dates to be entered without notifying the user is a poor UX and we will not add this feature to the library

**Reply by kevinbeal** [reply]
Okay. I'll be using it either way, but it changes the value without notifying the user. I share your concern and thought it should work differently for exactly that reason. 

It is a liability concern for certain applications that a form will change the values. We could be sued if we used the current built in behavior.

**Reply by rtivital** [reply]
If your application is so strict:
- do not use `allowFreeInput` to force user to pick a specific date
- do not use `minDate` and perform validation on your side

**Reply by kevinbeal** [reply]
That's reasonable. It conflicts with our best practices though. We have a policy of allowing the user to input any value and then handling validation through error messages rather than literally preventing the wrong values. Having the invalid values provides a history, allows them to be saved, but not processed. 

I've been building legal forms for over a decade. You can build your UI kit however you want, but it's not unheard of to build forms this way. 

I will be removing allowFreeInput until I can figure out some workaround. Or I will have the min date be in the day styling only, rather than use minDate prop.

I can make it work. I just thought it would make sense for fixOnBlur to work differently. Clearly you do not agree. That's fine. Thanks for the library.
